webapp/loggedin/modules/data/records/recordsView.js

```diff
@@ -17,19 +19,25 @@ import * as RecordsState from './recordsState'
 import * as SurveyState from '../../../../survey/surveyState'
 
 import { createRecord } from '../../../surveyViews/record/actions'
+import { initList } from '../../../tableViews/actions'
 
 const RecordsView = props => {
 
   const {
-    surveyInfo, user, nodeDefKeys, lang,
-    createRecord, history,
+    surveyInfo, surveyCycleKey, user, nodeDefKeys, lang,
+    createRecord, initList, history
   } = props
 
   const noCols = 3 + nodeDefKeys.length
   const gridTemplateColumns = `70px repeat(${noCols}, ${1 / noCols}fr) 50px 80px 80px 50px`
 
   const onRowClick = record => history.push(`${appModuleUri(dataModules.record)}${Record.getUuid(record)}`)
 
+  useOnUpdate(() => {
+    const moduleApiUri = `/api/survey/${Survey.getIdSurveyInfo(surveyInfo)}/${RecordsState.keys.records}`
```

Comment:
we need to find a better way to get the moduleApiUri. This is a duplication from TableView

Reply:
I'll associate the module to the state